main.go

```diff
@@ -82,14 +82,37 @@ func main() {
 		log.Fatalf(usage)
 	}
 
-	url, err := url.Parse(args[0])
+	rawURL := normalizeURL(args[0])
+	url, err := url.Parse(rawURL)
 	if err != nil {
 		log.Fatalf("could not parse url %q: %v", args[0], err)
 	}
 
 	visit(url)
 }
 
+// normalizeURL normalizes the url input.
+// If the scheme is missing, we assume it's "https" (or "http" if port 80).
+// If the host is missing, we assume it's "localhost".
+func normalizeURL(rawURL string) string {
+	if strings.Index(rawURL, "://") > -1 {
+		return rawURL
+	}
+
+	hostport := strings.Split(rawURL, ":")
```

Comment:
You are correct but using `net.SplitHostPort()` would break in some cases too since the input is the raw url string and not just the "host:port".